ecl/hql/hqlgram2.cpp

```diff
@@ -12337,15 +12338,15 @@ void parseAttribute(IHqlScope * scope, IFileContents * contents, HqlLookupContex
     OwnedHqlExpr parsed = scope->lookupSymbol(name, LSFsharedOK|LSFnoreport, ctx);
     ctx.incrementAttribsProcessed();
 
-    if (parsed && !alreadySimplified)
+    if (parsed && (!alreadySimplified || !cacheUptoDate || ctx.regenerateCache()))
```

Comment:
I don't think this condition needs to change.  if alreadySimplified is true then cacheUpToDate must be true and regenerate cache must be false.